sailor/_base/masterdata.py

```diff
@@ -0,0 +1,208 @@
+import warnings
+import logging
+from collections import Counter
+from collections.abc import Sequence
+from typing import Union
+
+import pandas as pd
+import plotnine as p9
+
+from ..utils.plot_helper import _default_plot_theme
+from ..utils.utils import _is_non_string_iterable
+
+LOG = logging.getLogger(__name__)
+LOG.addHandler(logging.NullHandler())
+
+
+class MasterDataField:
```

Comment:
this class should also be prefixed with `_` since nothing about it is meant to be public and all of the subclasses are private, too.

Reply:
well everything here is in a package that starts with an underscore (`_base`).
I don't really know what the conventions around this are, my assumption was that that should be enough to make it private?

Reply:
I don't think its enough. 
E.g. `MasterDataEntity` below is exposed to the public indirectly through the public methods on it. And that's correct because they have documentation on them that can be viewed also in our online docs. 
`MasterDataField` on the other hand is of no concern to the public. This class should also not appear in the documentation.  I think we should change that.

Independent of that particular issue but very related: I just noticed that we also do not generate any documentation for modules that start with `_`. So nothing in this file will be visible in the documentation (bad). We can change that behaviour likely with configuring sphinx-apidoc differently. But maybe not in this PR. Needs some experimentation. 

**However, to me this raises the question if we should actually have modules starting with `_` and then mix private and public content in there? Very confusing.**